todo_app/templates/index.html

```diff
@@ -7,12 +7,65 @@ <h1 class="display-4">To-Do App</h1>
     <p class="lead">Just another to-do app.</p>
   </div>
 
+  <div class="row mb-4">
+    <div class="col-md-12">
+      <h2>Add Item</h2>
+      <form action="{{ url_for('index') }}" method="post">
+          <div class="form-row">
+            <div class="form-group col-md-8">
+              <input type="text" class="form-control" name="title" placeholder="Enter a task here" required>
+            </div>
+            <div class="form-group col-md-3">
+              <button type="submit" class="btn btn-success btn-block">Submit</button>
+            </div>
+          </div>
+      </form>
+    
+      {% if error %}
+        <p class="text-danger">{{ error }}</p>
+      {% endif %}
+    </div>
+  </div>
+
   <div class="row">
     <div class="col-md-12">
       <h2>Items</h2>
-      <ul class="list-group mb-4">
-        <!-- List the todo items here -->
-      </ul>
+      <table class="table table-bordered mb-4">
+        <thead class="thead-dark">
+          <tr>
+            <th>#</th>
+            <th>Title</th>
+            <th>Status</th>
+            <th>Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for index in range(0, todos|length) %}
+              {% set todo = todos[index] %}
+              <tr>
+                <td>{{index + 1}}</td>
+                <td>{{todo.title}}</td>
+                <td>{{todo.status}}</td>
+
+                <td class="d-flex justify-content-center">
+
+                  {% if todo.status != 'Completed' %}
+                    <form action="{{ url_for('mark_completed') }}" method="post">
+                      <input type="hidden" name="todo_item" value='{{ todo|tojson }}'>
```

Comment:
Passing through the value as its entire JSON form makes things a little more involved - instead we'd usually just pass through the id & then complete or delete the item matching that id

Reply:
Cheers Jack, done.